sig/polyfill.rbs

```diff
@@ -11,24 +11,6 @@ module Gem
   end
 end
 
-class OptionParser
```

Comment:
OptionParser polyfill for internal is not necessary :put_litter_in_its_place: 

The new type definition of `OptionParser#on` introduces a problem. I guess it is a Steep's bug. see https://github.com/soutaro/steep/issues/389